stats/src/main/java/com/facebook/airlift/stats/cardinality/DenseHll.java

```diff
@@ -455,42 +429,54 @@ public DenseHll mergeWith(DenseHll other)
                     numberOfBuckets(other.indexBitLength)));
         }
 
-        int baseline = Math.max(this.baseline, other.baseline);
+        int newBaseline = Math.max(this.baseline, other.baseline);
         int baselineCount = 0;
 
-        int overflows = 0;
-        int[] overflowBuckets = new int[OVERFLOW_GROW_INCREMENT];
-        byte[] overflowValues = new byte[OVERFLOW_GROW_INCREMENT];
+        int bucket = 0;
+        for (int i = 0; i < deltas.length; i++) {
+            int newSlot = 0;
 
-        int numberOfBuckets = numberOfBuckets(indexBitLength);
-        for (int i = 0; i < numberOfBuckets; i++) {
-            int value = Math.max(getValue(i), other.getValue(i));
+            byte slot1 = deltas[i];
+            byte slot2 = other.deltas[i];
 
-            int delta = value - baseline;
-            if (delta == 0) {
-                baselineCount++;
-            }
-            else if (delta > MAX_DELTA) {
-                // grow overflows arrays if necessary
-                overflowBuckets = Ints.ensureCapacity(overflowBuckets, overflows + 1, OVERFLOW_GROW_INCREMENT);
-                overflowValues = Bytes.ensureCapacity(overflowValues, overflows + 1, OVERFLOW_GROW_INCREMENT);
+            for (int shift = 4; shift >= 0; shift -= 4) {
```

Comment:
Do you know why this number is 4? Is it because BITS_PER_BUCKET is 4 (I see that's used in calculating the size of the deltas array)?  If so, we should use the static variable explicitly here so they can't get out of sync.

Reply:
That makes sense - but ideally we should keep most of the code in sync with `airlift/airlift` so its easier to copy in future as well. So we can change there first and then copy it. Wdyt?

Reply:
that's fine with me.